From 02c2a067a5e76bb315d56e406313a1a2ac4c0948 Mon Sep 17 00:00:00 2001
From: Janne Pulkkinen <janne.pulkkinen@protonmail.com>
Date: Sat, 26 Aug 2023 15:42:00 +0300
Subject: [PATCH] Revert "Launch bwrap launcher using D-Bus instead of sock"

This is required for the Flatpak version of Protontricks, which does not
work without granting full access to D-Bus session bus with the new
implementation. Namely, without full access the `ProcessExited` signal
is not emitted for some reason, causing any Wine process to hang when
bwrap is in use.

Revert to the earlier deprecated `--socket` option. This allows bwrap to
work with the Flatpak version of Protontricks.

Refs #164

This reverts commit 7b333c6fd10da0fb1e7f51ed3701ba01361ea593.
---
 CHANGELOG.md                                    | 3 ---
 src/protontricks/data/scripts/bwrap_launcher.sh | 2 +-
 src/protontricks/data/scripts/wine_launch.sh    | 7 +------
 3 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/CHANGELOG.md b/CHANGELOG.md
index 3482155..7ea16bf 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -46,9 +46,6 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## [1.9.1] - 2022-08-28
 ### Added
  - Print a warning when multiple Steam directories are detected and `STEAM_DIR` is not used to specify the directory
- 
-### Changed
- - Launch Steam Runtime sandbox with `--bus-name` parameter instead of the now deprecated `--socket`
 
 ### Fixed
  - Fix various crashes due to Wine processes under Steam Runtime sandbox using the incorrect working directory
diff --git a/src/protontricks/data/scripts/bwrap_launcher.sh b/src/protontricks/data/scripts/bwrap_launcher.sh
index 6c9b975..3f07170 100644
--- a/src/protontricks/data/scripts/bwrap_launcher.sh
+++ b/src/protontricks/data/scripts/bwrap_launcher.sh
@@ -86,4 +86,4 @@ status_fd="$1"
 
 exec "$STEAM_RUNTIME_PATH"/run --share-pid --launcher --pass-fd "$status_fd" \
 "${mount_params[@]}" -- \
---info-fd "$status_fd" --bus-name="com.github.Matoking.protontricks.App${STEAM_APPID}_${PROTONTRICKS_SESSION_ID}"
+--info-fd "$status_fd" --socket="$PROTONTRICKS_TEMP_PATH/launcher.sock"
diff --git a/src/protontricks/data/scripts/wine_launch.sh b/src/protontricks/data/scripts/wine_launch.sh
index 4511a18..18b3b29 100644
--- a/src/protontricks/data/scripts/wine_launch.sh
+++ b/src/protontricks/data/scripts/wine_launch.sh
@@ -168,11 +168,6 @@ elif [[ "$PROTONTRICKS_STEAM_RUNTIME" = "bwrap" ]]; then
 
     log_info "Starting Wine process using 'pressure-vessel-launch'"
 
-    # It would be nicer to use the PID here, but that would break multiple
-    # simultaneous Protontricks sessions inside Flatpak, which doesn't seem to
-    # expose the unique host PID.
-    bus_name="com.github.Matoking.protontricks.App${STEAM_APPID}_${PROTONTRICKS_SESSION_ID}"
-
     # Pass all environment variables to 'steam-runtime-launch-client' except
     # for problematic variables that should be determined by the launch command
     # instead.
@@ -189,7 +184,7 @@ elif [[ "$PROTONTRICKS_STEAM_RUNTIME" = "bwrap" ]]; then
     done
 
     exec "$STEAM_RUNTIME_LAUNCH_SCRIPT" \
-    --share-pids --bus-name="$bus_name" \
+    --share-pids --socket="$PROTONTRICKS_TEMP_PATH/launcher.sock" \
     --directory "$PWD" \
     --env=PROTONTRICKS_INSIDE_STEAM_RUNTIME=1 \
     "${env_params[@]}" -- "$PROTONTRICKS_PROXY_SCRIPT_PATH" "$@"
-- 
2.40.1

